### PR TITLE
[Hub Generated] Review request for Microsoft.Insights to add version preview/2021-03-03-preview

### PR DESCRIPTION
**Pull request by xiaomi7732**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/xiaomi7732/azure-rest-api-specs/dev-applicationinsights-Microsoft.Insights-2021-03-03-preview...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by jianyexi** [comment]
@xiaomi7732  For below error, it's because the swaggers under  package-preview-2020-06 contains two operation 'Operation_List'  definitions , but since these errors are not introduced by this PR, so you can choose to fix it or not 

image

**Comment by xiaomi7732** [comment]
Thanks @jianyexi for the explaination. @kairu-ms, could you please review this PR?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    }
+  },
+  "paths": {
+    "/{resourceUri}/providers/Microsoft.Insights/generateDiagnosticServiceReadOnlyToken": {
+      "post": {
+        "tags": [
+          "generateDiagnosticServiceReadOnlyToken"
+        ],
+        "operationId": "GenerateDiagnosticServiceReadOnlyToken",
+        "description": "**Gets an read-only access token for application insights diagnostic service data.**",
```
Comment (kairu-ms) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json:43]: Can you remove "**"

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      }
+    }
+  },
+  "paths": {
+    "/{resourceUri}/providers/Microsoft.Insights/generateDiagnosticServiceReadOnlyToken": {
+      "post": {
+        "tags": [
+          "generateDiagnosticServiceReadOnlyToken"
+        ],
+        "operationId": "GenerateDiagnosticServiceReadOnlyToken",
```
Comment (kairu-ms) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json:42]: OperationId should contain two parts and seperated by "_". For example:

```
"operationId": "DiagnosticServiceToken_GenerateReadOnly"
```

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        }
+      }
+    },
+    "/{resourceUri}/providers/Microsoft.Insights/generateDiagnosticServiceReadWriteToken": {
+      "post": {
+        "tags": [
+          "generateDiagnosticServiceReadWriteToken"
+        ],
+        "operationId": "GenerateDiagnosticServiceReadWriteToken",
+        "description": "**Gets an read-write access token for application insights diagnostic service data.**",
```
Comment (kairu-ms) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json:84]: Remove "**"

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+          }
+        }
+      }
+    },
+    "/{resourceUri}/providers/Microsoft.Insights/generateDiagnosticServiceReadWriteToken": {
+      "post": {
+        "tags": [
+          "generateDiagnosticServiceReadWriteToken"
+        ],
+        "operationId": "GenerateDiagnosticServiceReadWriteToken",
```
Comment (kairu-ms) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json:83]:
```
"operationId": "DiagnosticServiceToken_GenerateReadWrite"
```

File: specification/applicationinsights/resource-manager/readme.md [hunk 5]
```diff
@@ -251,7 +264,7 @@ input-file:
 
 These settings apply only when `--tag=package-preview-2020-10` is specified on the command line.
 
-```yaml $(tag) == 'package-preview-2020-10'
+``` yaml $(tag) == 'package-preview-2020-10'
```
Comment (kairu-ms) [inline specification/applicationinsights/resource-manager/readme.md:267]:
```suggestion
```yaml $(tag) == 'package-preview-2020-10'
```

File: specification/applicationinsights/resource-manager/readme.md [hunk 6]
```diff
@@ -28,7 +28,7 @@ These are the global settings for the ApplicationInsights API.
 title: ApplicationInsightsManagementClient
 description: Composite Swagger for Application Insights Management Client
 openapi-type: arm
-tag: package-preview-2020-10
+tag: package-preview-2021-03
```
Comment (kairu-ms) [inline specification/applicationinsights/resource-manager/readme.md:31]: If you want to use tag `package-preview-2021-03`, you should create a tag `package-preview-2021-03` which inherits from `package-preview-2020-10`. Inherit means to copy all input-files from `package-preview-2020-10`.

Comment (xiaomi7732) [reply specification/applicationinsights/resource-manager/readme.md:31]: I'll revert the change to `package-preview-2020-10`.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json (hunk 4 above)
Comment (xiaomi7732) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json]: Done

File: specification/applicationinsights/resource-manager/readme.md (hunk 5 above)
Comment (xiaomi7732) [reply specification/applicationinsights/resource-manager/readme.md]: Done.

**Comment by xiaomi7732** [comment]
@kairu-ms Thank you for the review. I've updated the authoring according the feedback. Please review it again. Thank you!

File: specification/applicationinsights/resource-manager/readme.md (hunk 6 above)
Comment (kairu-ms) [reply specification/applicationinsights/resource-manager/readme.md]: OK. BTW the tag `package-preview-2020-10` is invalid. I will change it to `package-2020-03` later.

**Review by kairu-ms — APPROVED** [review]

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/examples/DiagnosticServicesReadOnlyTokenPost.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
+  "parameters": {
+    "resourceUri": "subscriptions/df602c9c-7aa0-407d-a6fb-eb20c8bd1192/resourceGroups/FabrikamFiberApp/providers/microsoft.insights/components/CustomAvailabilityTest/providers/Microsoft.Insights/generateDiagnosticServiceReadOnlyToken",
```
Comment (varunkumta) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/examples/DiagnosticServicesReadOnlyTokenPost.json:3]: >providers/Microsoft.Insights/generateDiagnosticServiceReadOnlyToken [](start = 166, length = 67)

remove this segment of the url

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful response containing the application insights diagnostic services token.",
+            "schema": {
+              "$ref": "#/definitions/DiagnosticServicesTokenResponse"
+            },
+            "examples": {
```
Comment (varunkumta) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json:64]: >examples [](start = 13, length = 8)

I don't think you need to provide examples here .. the x-ms-examples should be enough

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/examples/DiagnosticServicesReadOnlyTokenPost.json (hunk 7 above)
Comment (xiaomi7732) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/examples/DiagnosticServicesReadOnlyTokenPost.json]: Good to know. Fixed

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json (hunk 8 above)
Comment (xiaomi7732) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2021-03-03-preview/diagnosticServicesToken_API.json]: Good to know. Fixed

**Comment by xiaomi7732** [comment]
@PhoenixHe-msft / @jianyexi thanks for looking into it. It was mentioned that those errors were not introduced by this change. And I am intended to keep this change as minimal. Is it possible to have a separate effort to fix those?

**Comment by msyyc** [comment]
To fix failure for python, please change 'v2020_06_02-preview' to 'v2020_06_02_preview' in 'readme.python.md'

image

**Comment by kairu-ms** [comment]
> fix failure for python

It will be fixed in this PR https://github.com/Azure/azure-rest-api-specs/pull/13875

**Pull request merged**